From 02eb3ec5a98e0e5f92f9f8fdc9c5142dc80bc217 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Tue, 14 Jan 2014 15:40:16 +0100
Subject: [PATCH 239/267] coredumpctl: in case of error free pattern after
 print

(cherry picked from commit cf50a55277ee432e1217ad894afcbd9a664ff96c)
---
 src/journal/coredumpctl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/journal/coredumpctl.c b/src/journal/coredumpctl.c
index 560a91b..a5997e2 100644
--- a/src/journal/coredumpctl.c
+++ b/src/journal/coredumpctl.c
@@ -126,10 +126,11 @@ static int add_match(Set *set, const char *match) {
                 goto fail;
 
         log_debug("Adding pattern: %s", pattern);
-        r = set_consume(set, pattern);
+        r = set_put(set, pattern);
         if (r < 0) {
                 log_error("Failed to add pattern '%s': %s",
                           pattern, strerror(-r));
+                free(pattern);
                 goto fail;
         }
 
-- 
1.8.4.5

